app/src/main/java/ch/epfl/sdp/appart/user/UserViewModel.java

```diff
@@ -53,8 +58,52 @@ public void putUser(User user) {
      * @param user the user to update in database
      */
     public void updateUser(User user) {
-        CompletableFuture<Boolean> updateUser = db.updateUser(user, mUri.getValue());
-        updateUser.thenAccept(mUpdateCardConfirmed::setValue);
+        CompletableFuture<Boolean> updateUser = db.updateUser(user);
+        updateUser.exceptionally(e -> {
+            Log.d("UPDATE USER", "DATABASE FAIL");
+            return null;
+        });
```

Comment:
same as above. Also, I'm not sure I understand when `putUser` is used over `updateUser` inside the USerViewModel. If we have access to the user page, it means there already is a user entry on firestore, right?

Reply:
PutUser is actually never used at the moment (only the mock implementation in MockDatabaseService is actually being used). This is because the user gets put the first time once the account is created and thus from there on it is a continuous **update**. Good point, we are not actually ever using this (as part of user view model). We could use it to store the user in the first place instead of directly calling the database methods or simply just delete this method. We can decide on this next time maybe!